esmvaltool/diag_scripts/ocean/diagnostic_tools.py

```diff
@@ -225,22 +338,40 @@ def make_cube_layer_dict(cube):
     """
     #####
     # Check layering:
-    depth = cube.coords('depth')
+    coords = cube.coords()
+    layers = []
+    for coord in coords:
+        if coord.standard_name in ['depth', 'region']:
+            layers.append(coord)
+
     cubes = {}
+    if layers == []:
+        cubes[''] = cube
+        return cubes
 
-    if depth == []:
+    if len(layers) > 1:
+        # This field has a strange number of layer dimensuions.
+        # depth and regions?
+        assert 0
```

Comment:
raise a ValueError instead of using assert

Reply:
Done!